solr/core/src/java/org/apache/solr/update/DirectUpdateHandler2.java

```diff
@@ -975,6 +976,14 @@ private void updateDocOrDocValues(AddUpdateCommand cmd, IndexWriter writer) thro
     }
   }
 
+  boolean forciblyAddBlockId() {
```

Comment:
this is here in part so that we can test _not_ doing this (i.e. current behavior)... but I don't think it's worth it.  If we want to _not_ do it conditionally then it could be via luceneMatchVersion but I don't care either way as it's minor to add it unconditionally.

Reply:
The change described above causes BlockUpdateTest#testLegacyBlockProcessing to fail.
Should I remove it, I noticed you mentioned this being a burden going forward, or should I create a new solrconfig and test case to test this works in older solr versions?
Wouldn't this be a problem if _root_ is not defined in the collection's schema?

Reply:
Come to think of it, who cares if root is populated or not for "old" schemas?  Lets just do it always; no conditional test.  If someone doesn't want root to be populated, that same person probably doesn't have child docs and should just as well omit the root field from their schema.